dandi/tests/test_validate.py

```diff
@@ -1,35 +1,62 @@
 from glob import glob
 import os
 
-import appdirs
+# import appdirs
+import pytest
 
+from .fixtures import BIDS_TESTDATA_SELECTION
 
-def test_validate_bids(bids_examples, tmp_path):
+
+@pytest.mark.parametrize("dataset", BIDS_TESTDATA_SELECTION)
+def test_validate_bids(bids_examples, tmp_path, dataset):
     from ..validate import validate_bids
 
-    # TEST DEFAULT REPORT PATH:
-    # Replace explicit dataset with `os.listdir(bids_examples)[1]` whenever we
-    # implement light data cloning analogous to (`[1]` because `[0]` is .git):
-    # https://github.com/bids-standard/bids-specification/pull/1143
-    selected_dataset = os.path.join(bids_examples, "asl003")
-    _ = validate_bids(selected_dataset, report=True)
+    selected_dataset = os.path.join(bids_examples, dataset)
+    validation_result = validate_bids(selected_dataset, report=True)
+    for i in validation_result:
+        assert not hasattr(i, "severtiy")
+
+
+def test_report_path(bids_examples, tmp_path):
+    from ..validate import validate_bids
 
-    # Check if a report is being produced.
-    pid = os.getpid()
-    log_dir = appdirs.user_log_dir("dandi-cli", "dandi")
-    report_expression = os.path.join(log_dir, f"bids-validator-report_*-{pid}.log")
-    assert len(glob(report_expression)) == 1
-
-    # TEST CISTOM REPORT PATH:
-    # Replace explicit dataset with `os.listdir(bids_examples)[1]` whenever we
-    # implement light data cloning analogous to (`[1]` because `[0]` is .git):
-    # https://github.com/bids-standard/bids-specification/pull/1143
     report_path = os.path.join(tmp_path, "inplace_bids-validator-report.log")
-    selected_dataset = os.path.join(bids_examples, "asl003")
+    selected_dataset = os.path.join(bids_examples, BIDS_TESTDATA_SELECTION[0])
     _ = validate_bids(
         selected_dataset,
         report_path=report_path,
     )
 
     # Check if a report is being produced.
     assert len(glob(report_path)) == 1
+
+
+@pytest.mark.parametrize(
+    "dataset", ["invalid_asl003", "invalid_eeg_cbm", "invalid_pet001"]
+)
+def test_validate_bids_errors(bids_error_examples, dataset):
+    # This only checks that the error we found is correct, not that we found all errors.
+    # ideally make a list and erode etc.
+    import json
+    import pathlib
+
+    from ..validate import validate_bids
```

Comment:
I see no reason to place these imports inside the function instead of at file scope.